src/toil/jobStores/azureJobStore.py

```diff
@@ -73,61 +75,215 @@ def _fetchAzureAccountKey(accountName):
+    # URLs where the may interfere with the certificate common name. We use a double
+    # underscore as a separator instead.
+    #
+    # See https://msdn.microsoft.com/en-us/library/azure/dd135715.aspx
 
-    # Table names must be alphanumeric
-    nameSeparator = 'xx'
+    containerNameRe = re.compile(r'^[a-z0-9](-?[a-z0-9]+)+[a-z0-9]$')
+    minContainerNameLen = 3
+    maxContainerNameLen = 63
+    maxNameLen = 10
+    nameSeparator = 'xx'  # Table names must be alphanumeric
 
     # Length of a jobID - used to test if a stats file has been read already or not
     jobIDLength = len(str(uuid.uuid4()))
 
-    def qualify(self, name):
-        return self.namePrefix + self.nameSeparator + name
+    @classmethod
+    def createJobStore(cls, locator, config, **kwargs):
+        cls._checkJobStoreCreation(create=True, locator=locator)
+        info = cls._processLocator(locator)
+        jobStore = cls(accountName=info.accountName, accountKey=info.accountKey, namePrefix=info.namePrefix,
+                       tableService=info.tableService, blobService=info.blobService,
+                       jobItems=cls._createTable(info.tableService, cls._qualify('jobs', info.namePrefix)),
+                       jobFileIDs=cls._createTable(info.tableService, cls._qualify('jobFileIDs', info.namePrefix)),
+                       files=cls._createBlobContainer(info.blobService, cls._qualify('files', info.namePrefix)),
+                       statsFiles=cls._createBlobContainer(info.blobService, cls._qualify('statsfiles',
+                                                                                          info.namePrefix)),
+                       statsFileIDs=cls._createTable(info.tableService, cls._qualify('statsFileIDs', info.namePrefix)),
+                       config=config, **kwargs)
+
+        jobStore._createJobStore(config)
+        if jobStore.config.cseKey is not None:
+            jobStore.keyPath = jobStore.config.cseKey
+        return jobStore
+
+    @classmethod
+    def loadJobStore(cls, locator, **kwargs):
+        cls._checkJobStoreCreation(create=False, locator=locator)
+        info = cls._processLocator(locator)
+        jobStore = cls(accountName=info.accountName, accountKey=info.accountKey, namePrefix=info.namePrefix,
+                       tableService=info.tableService, blobService=info.blobService,
+                       jobItems=cls._getTable(info.tableService, cls._qualify('jobs', info.namePrefix)),
+                       jobFileIDs=cls._getTable(info.tableService, cls._qualify('jobFileIDs', info.namePrefix)),
+                       files=cls._getBlobContainer(info.blobService, cls._qualify('files', info.namePrefix)),
+                       statsFiles=cls._getBlobContainer(info.blobService, cls._qualify('statsfiles', info.namePrefix)),
+                       statsFileIDs=cls._getTable(info.tableService, cls._qualify('statsFileIDs', info.namePrefix)),
+                       config=None, **kwargs)
```

Comment:
This looks like it is exactly the same as in the method above. Can't we push a lot of this into the constructor?